### ISSUE TITLE
feature request: add ability to restart apps from applications list page in ArgoCD

### ISSUE THREAD

**Issue by audip** [argoproj/argo-cd#21305]
# Summary

A button to "Restart Apps" on the applications list page with the ability to restart selected apps.

# Motivation

There are use-cases of a central ArgoCD cluster deploying to Kubernetes clusters across multiple AZs/regions around the world that need to be restarted when a cache invalidation event or another maintenance event happens. Rolling restart is an action that engineers at @myfitnesspal perform when pods aren't able to pick new IP address of Elasticache instances, actively performing incident response. So, I'd like to propose adding a button that we are using internally, see screenshots of how it looks and operates (very similar to the "Refresh Apps" button)

![image](https://github.com/user-attachments/assets/d899c9be-1c32-4822-bde6-f0c3e299e0ea)

![image](https://github.com/user-attachments/assets/fa8dc54e-9cae-4123-8704-9a720fe7fdb9)

# Proposal

The code is about ~100 lines as it is mostly a UI change with functionality to perform a rolling restart already existing. The source code is available for viewing in this pull request: https://github.com/argoproj/argo-cd/pull/21307

**Comment by andrii-korotkov-verkada** [comment]
What's the semantic of restarting apps? Does it do rollout restart on all deployments and rollouts? How does it handle job and cronjobs running?

**Comment by audip** [comment]
> What's the semantic of restarting apps? Does it do rollout restart on all deployments and rollouts? How does it handle job and cronjobs running?

Hello Andrii, thanks for writing back! It primarily focuses on rollout restart of deployments. We aren't doing Argo Rollouts at scale.

I'm open to suggestions on how to handle jobs and cronjobs.

**Comment by agaudreault** [comment]
I don't think this is the kind of feature that can be added in Argo CD UI for all users. The semantic of restarting apps can be very different from one company to another, or one application to another. 

If you are maintaining a fork to add this feature, many UI features are using extensions. I think this work should can be aimed towards allowing Argo application list page buttons to be added via an extension, rather then adding this feature natively. This way, you do not have to maintain a fork for this feature at your company.

https://argo-cd.readthedocs.io/en/stable/developer-guide/extensions/ui-extensions/

I will close this issue, but feel free to open a new one for the UI extension enhancement if necessary.
